google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/Exceptions.java

```diff
@@ -106,12 +141,32 @@ public static StorageException toStorageException(
     if (error == null) {
       return null;
     }
+    String streamName = error.getEntity();
+    String errorMessage =
+        error.getErrorMessage().indexOf("Entity") > 0
+            ? error.getErrorMessage().substring(0, error.getErrorMessage().indexOf("Entity")).trim()
+            : error.getErrorMessage().trim();
+    // Note: if errorMessage format changes, the parsing logic below might break
+    Long expectedOffet =
+        Long.parseLong(
+            errorMessage.substring(
+                errorMessage.lastIndexOf("offset") + 7, errorMessage.lastIndexOf(",")));
```

Comment:
Have a whole stream match pattern for "expected offset $0, received $1". I am writing cl/445470232 to make sure the parsing will not break.

Reply:
Done.